### PR TITLE
DDF-4877 Adding a null check for metacard property

### PR DESCRIPTION
**Pull request by nsuvarna**
#### What does this PR do?
Adds a null check to prevent a console error when using a form with unknown attributes.
#### Who is reviewing it? 

@Bdthomson 
@Schachte 
@gordocanchola 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@bdeining 
@troymohl 

#### How should this be tested?

1. Add these files to `etc/forms`: [test-form.zip](https://github.com/codice/ddf/files/3218400/test-form.zip)
2. In the karaf console, execute `forms:load`
3. In intrigue, create a query using the form `Test Form` and run it
4. Ensure there are no console errors
#### What are the relevant tickets?
For GH Issues:
Fixes: #4877 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/js/CQLUtils.js [hunk 1]
```diff
@@ -237,6 +237,9 @@ function generateFilter(type, property, value, metacardDefinitions) {
   if (!metacardDefinitions) {
     metacardDefinitions = require('../component/singletons/metacard-definitions.js')
   }
+  if (metacardDefinitions.metacardTypes[property] === undefined) {
+    return null
```
Comment (rzwiefel) [inline ui/packages/catalog-ui-search/src/main/webapp/js/CQLUtils.js]: is it safe to return null here? None of the callers will break?

Comment (nsuvarna) [reply ui/packages/catalog-ui-search/src/main/webapp/js/CQLUtils.js]: I believe that the only way to actually query on a property that is unknown/undefined is by using an imported search form, and I didn't notice any issues with that after making this change. Also, from what I can tell, `filter-builder` builds the cql using `CQLUtil`'s `transformFilterToCQL` function, which uses `cql.js`'s read/write functions that can handle null filters.

**Comment by Bdthomson** [comment]
build now

**Comment by Bdthomson** [comment]
✅ Hero Successful
Verified:
- There is no longer a UI breaking JS error when a search form is loaded with an attribute that the UI is not aware of.

**Pull request merged**